docs/contributing/upgrading-dependencies.md

```diff
@@ -0,0 +1,121 @@
+
+### How is this dependency used in the project?
+
+  - Core dependencies like Electron, React and Typescript are critical to
+    Desktop - without these dependencies we'd have a very different product
+  - Application dependencies are other external packages which are needed at
+    runtime - how can we test they still work as expected?
+  - Tooling dependencies help to build and verify Desktop works as expected, but
+    aren't shipped to users
+
+### If we ship a bad update due to updating a dependency, what is the impact to users?
+
+  - If a core dependency is broken, it may cause the app to crash, lose the
+    user's work, or require the user to reinstall an old version of Desktop to
+    get back to a usable state
+  - An application dependency breaking may affect some user's workflows and
+    their ability to work
+  - Broken tooling impacts the developers and their ability to work on the
+    project, and potentially affects shipping builds to users, but otherwise
+    might not be seen
+
+### Can we leverage automation to verify a change is safe?
+
+  - Core dependencies need to be tested in a variety of setups due to their
+    complexity, and may still require manual testing in some cases even with
+    automated tests in place
+  - Application dependencies may be able to be wrapped in tests, to help
+    identify regressions before a change is released to users
+  - Tooling dependencies are used heavily as part of continuous integration
+    tests, and any potential problems will be caught early
+
+## Grouping dependencies by impact
+
+With the criteria above, we can group dependencies into these three buckets that
+help guide when they should be updated.
+
+### Core dependencies
+
+These are the most important dependencies to the app, and include:
+
+ - `package.json`
+   - `@types/node`
+   - `electron`
+   - `electron-builder`
+   - `electron-packager`
+   - `electron-winstaller`
+   - `typescript`
+   - `webpack` and related dependencies
+ - `app/package.json`
+   - `codemirror`
+   - `dugite`
+   - `react`
+   - `react-dom`
+   - `keytar`
+
+Anyone who wants to get involved with updating these versions should be very
+familiar with both the library and what has changed between versions, so the
+review process is focused on verifying the impact of the change rather than
+understanding what has changed.
+
+The maintainers have significant experience with these dependencies, and may
+wish to upgrade one of these dependencies at a later time than when a
+contribution is opened by the community.
+
+### Application dependencies
+
+This group of dependencies represent the rest of the listed dependencies in
+`app/pacakge.json` - there are lots of small dependencies which are approachable
+for external contributions.
+
+If you are interested in upgrading one of these, take a look at where it is
+used in the application and confirm the affected functionality works as
+expected before opening a pull request.
```

Comment:
Is it work mentioning about the "test" releases here i.e. `1.4.1-test1`?

Reply:
I don't think so. I consider those test releases special snowflake builds that we might use to verify a specific fix (it's often easier to link an installer to someone than to get them to work with the source) but beyond that I consider them unofficial.